helpers/mathjax.js

```diff
@@ -61,35 +61,19 @@ export class HtmlBlockMathRenderer {
 		// This work-around should be removed when linebreaks are natively supported.
 		// MathJax issue: https://github.com/mathjax/MathJax/issues/2312
 		// A duplicate that explains our exact issue: https://github.com/mathjax/MathJax/issues/2495
-		const lineBreakStyle = 'display: block; height: 0.5rem;';
-
-		// If we're opting out of deferred rendering, we need to rely
-		// on the global MathJax install for rendering.
-		if (options.noDeferredRendering) {
-			elem.querySelectorAll('mspace[linebreak="newline"]').forEach(elm => {
-				elm.setAttribute('style', lineBreakStyle);
-			});
-
-			await window.MathJax.startup.promise;
-			renderingPromise = renderingPromise.then(() => window.MathJax.typesetShadow(elem.getRootNode(), elem));
-			await renderingPromise;
-			return elem;
-		}
-
-		const inner = elem.innerHTML.replace(/<mspace linebreak="newline">/gi, `<mspace linebreak="newline" style="${lineBreakStyle}">`);
-
-		const temp = document.createElement('div');
-		temp.style.display = 'none';
-		temp.attachShadow({ mode: 'open' });
-		temp.shadowRoot.innerHTML = `<div><mjx-doc><mjx-head></mjx-head><mjx-body>${inner}</mjx-body></mjx-doc></div>`;
+		elem.querySelectorAll('mspace[linebreak="newline"]').forEach(elm => {
+			elm.setAttribute('style', 'display: block; height: 0.5rem;');
+		});
 
-		elem.appendChild(temp);
+		// If we're using deferred rendering, we need to create a document structure
+		// within the element so MathJax can appropriately process math.
+		if (!options.noDeferredRendering) elem.innerHTML = `<mjx-doc><mjx-head></mjx-head><mjx-body>${elem.innerHTML}</mjx-body></mjx-doc>`;
 
 		await window.MathJax.startup.promise;
-		renderingPromise = renderingPromise.then(() => window.MathJax.typesetShadow(temp.shadowRoot));
+		renderingPromise = renderingPromise.then(() => window.MathJax.typesetShadow(elem.getRootNode(), elem));
 		await renderingPromise;
 
-		return temp.shadowRoot.firstChild;
+		return elem;
```

Comment:
If I follow this correctly, this now returns the same element that was passed in. Is that correct?

Reply:
That's right, yeah.

Reply:
Technically I suppose that makes the return value completely pointless, since the reference to it in the html-block rendering code will just implicitly update.

Reply:
Yeah, and also a point of failure as we've been chatting about.  If all the renderers currently return the element it was passed, maybe `d2l-html-block` shouldn't expect it to be returned, and instead assume that all processing will be done on the element it passes. Not sure if this would be limiting in the future, as the previous version of MathJax renderer required the ability to return a new element. 🤔

Reply:
Yeah, that was my main thought. Certainly right now the existing renderers act on the elements as they are, and enforcing that in the component makes it very obvious that we're expecting renderers to run inline. I don't think it's particularly difficult to reverse course later if we want. Realistically we would just switch the renderers to start returning the element and then adjust the html-block to start consuming it again. I'm certainly fine removing the return value for now though.

Reply:
Ok, I made this change in the core renderers and the html-block component itself. It will need to be reflected in the @mentions renderer as well, but because that's in a different repo that change will have to wait until post-merge.

Reply:
Cool.  Thanks for updating that.  It would be quite unfortunate if a plugin renderer forgot to return the element and in effect broken the component or other renderers.

Reply:
Haha yeah, I was trying to think if any renderer could have a valid use case for wanting to return something, but realistically if a renderer needs to detach the element from the DOM or something, it's also going to need to add it back as well. At least this way the requirement is super clear.